extensions/blocks/videopress/edit.js

```diff
@@ -90,52 +109,158 @@ const VideoPressEdit = CoreVideoEdit =>
+				? __( 'Note: Autoplaying videos may cause usability issues for some visitors.', 'jetpack' )
+				: null;
+		};
+
 		render() {
 			const {
 				attributes,
 				className,
+				instanceId,
 				isFetchingPreview,
 				isSelected,
 				isUploading,
 				preview,
 				setAttributes,
 			} = this.props;
 			const { fallback, isFetchingMedia } = this.state;
+			const { autoplay, caption, controls, loop, muted, poster, preload } = attributes;
+
+			const videoPosterDescription = `video-block__poster-image-description-${ instanceId }`;
+
+			const blockSettings = (
+				<Fragment>
+					<BlockControls>
+						<Toolbar>
+							<IconButton
+								className="components-icon-button components-toolbar__control"
+								label={ __( 'Edit video', 'jetpack' ) }
+								onClick={ this.switchToEditing }
+								icon="edit"
+							/>
+						</Toolbar>
+					</BlockControls>
+					<InspectorControls>
+						<PanelBody title={ __( 'Video Settings', 'jetpack' ) }>
+							<ToggleControl
+								label={ __( 'Autoplay', 'jetpack' ) }
+								onChange={ this.toggleAttribute( 'autoplay' ) }
+								checked={ autoplay }
+								help={ this.getAutoplayHelp }
+							/>
+							<ToggleControl
+								label={ __( 'Loop', 'jetpack' ) }
+								onChange={ this.toggleAttribute( 'loop' ) }
+								checked={ loop }
+							/>
+							<ToggleControl
+								label={ __( 'Muted', 'jetpack' ) }
+								onChange={ this.toggleAttribute( 'muted' ) }
+								checked={ muted }
+							/>
+							<ToggleControl
+								label={ __( 'Playback Controls', 'jetpack' ) }
+								onChange={ this.toggleAttribute( 'controls' ) }
+								checked={ controls }
+							/>
+							<SelectControl
+								label={ __( 'Preload', 'jetpack' ) }
+								value={ preload }
+								onChange={ value => setAttributes( { preload: value } ) }
+								options={ [
+									{ value: 'auto', label: __( 'Auto', 'jetpack' ) },
+									{ value: 'metadata', label: __( 'Metadata', 'jetpack' ) },
+									{ value: 'none', label: __( 'None', 'jetpack' ) },
+								] }
+							/>
+							<MediaUploadCheck>
+								<BaseControl
+									className="editor-video-poster-control"
+									label={ __( 'Poster Image', 'jetpack' ) }
+								>
+									<MediaUpload
+										title={ __( 'Select Poster Image', 'jetpack' ) }
+										onSelect={ this.onSelectPoster }
+										allowedTypes={ VIDEO_POSTER_ALLOWED_MEDIA_TYPES }
+										render={ ( { open } ) => (
+											<Button
+												isDefault
+												onClick={ open }
+												ref={ this.posterImageButton }
+												aria-describedby={ videoPosterDescription }
+											>
+												{ ! poster
+													? __( 'Select Poster Image', 'jetpack' )
+													: __( 'Replace image', 'jetpack' ) }
+											</Button>
+										) }
+									/>
+									<p id={ videoPosterDescription } hidden>
+										{ poster
+											? sprintf( __( 'The current poster image url is %s', 'jetpack' ), poster )
```

Comment:
Does `poster` contain a blob URL while the uploaded image is loading? Can that be confusing?

Reply:
> Does `poster` contain a blob URL while the uploaded image is loading?

Nope, the media modal prevent the image from being selected if it has not been uploaded yet.